crosshair/condition_parser.py

```diff
@@ -806,10 +806,89 @@ def get_class_invariants(self, cls: type) -> List[ConditionExpr]:
         return []
 
 
+class GivenParser(ConcreteConditionParser):
+    def __init__(self, toplevel_parser: ConditionParser = None):
+        super().__init__(toplevel_parser)
+
+    @staticmethod
+    def is_string_literal(node: ast.AST) -> bool:
+        if sys.version_info >= (3, 8):
+            return (
+                    isinstance(node, ast.Expr)
+                    and isinstance(node.value, ast.Constant)
+                    and isinstance(node.value.value, str)
+            )
+        else:
+            return isinstance(node, ast.Expr) and isinstance(node.value, ast.Str)
```

Comment:
No need to define these methods, as we'll inherit them from the parent class.

Reply:
Oh yeah i missed this, good catch.